### PR TITLE
KAFKA-12200 Migrate connect:file module to JUnit 5

### PR DESCRIPTION
**Pull request by tang7526**
https://issues.apache.org/jira/browse/KAFKA-12200

replace junit4 methods with junit5 methods

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

**Comment by ijuma** [comment]
Thanks for the PR. You also need to remove the project name from the `build.gradle` filter:

https://github.com/apache/kafka/blob/trunk/build.gradle#L245

File: build.gradle [hunk 1]
```diff
@@ -242,7 +242,7 @@ subprojects {
   }
 
   // Remove the relevant project name once it's converted to JUnit 5
-  def shouldUseJUnit5 = !(["api", "connect", "core", "file", "generator", "runtime", "examples",
+  def shouldUseJUnit5 = !(["api", "connect", "core", "generator", "runtime", "examples",
```
Comment (ijuma) [inline build.gradle:245]: You also have to update the JUnit dependency in the build, check one of the recent related commits for an example.

**Review by ijuma — APPROVED** [review]
LGTM, thanks.

**Comment by ijuma** [comment]
For some reason, the Jenkins job didn't trigger. Can you please merge trunk to this branch to see if that fixes it?

File: build.gradle [hunk 2]
```diff
@@ -2075,8 +2075,7 @@ project(':connect:file') {
     compile libs.slf4jApi
 
     testCompile libs.easymock
-    testCompile libs.junitJupiterApi
-    testCompile libs.junitVintageEngine
+    testCompile libs.junitJupiter
     testCompile libs.powermockJunit4
```
Comment (chia7712) [inline build.gradle]: It seems power mock is not used by this module anymore. Could we remove them?

**Review by chia7712 — APPROVED** [review]
@tang7526 Thanks for your nice patch. one trivial comment is left.

File: build.gradle (hunk 2 above)
Comment (tang7526) [reply build.gradle]: Done. They are already removed now.

**Pull request merged**